source/extensions/tracers/datadog/logger.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <datadog/logger.h>
+
+#include "source/common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+/**
+ * Logging adapter for use by dd-trace-cpp. This class implements dd-trace-cpp's
+ * datadog::tracing::Logger interface in terms of spdlog::logger, which is what
```

Comment:
non-actionable comment -- I'm just saying this into the wind...

It seems unfortunate that spdlog is a leaky abstraction from the Logger interface provided by Envoy's common library. I had thought that was an implementation detail, but I guess not.

This is some daunting tech-debt it would be nice to clean up some decade in the future.

Reply:
It is worth reconsidering the design I came up with. Was basically hacking around to see what worked.

The Envoy way is to use logging macros that conceal the underlying implementation. This has the advantage of allowing source file and line number to be included in the logged message.

We wouldn't be able to export those macros to a separately compiled library like dd-trace-cpp, and besides dd-trace-cpp has a policy of "no logging macros" for internal reasons :)

spdlog is a leaky abstraction, and regardless of what we use, dd-trace-cpp error messages will say they come from this component instead of from the underlying dd-trace-cpp component.

I'm open to ideas about how we can make dd-trace-cpp logging within Envoy more useful to users.